packages/provider-elasticsearch/src/example-types/results/highlighting/request.js

```diff
@@ -0,0 +1,108 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+import { CartesianProduct } from 'js-combinatorics'
+
+/**
+ * Return names for all fields groups and their sub-fields in `schema`.
+ */
+const getFieldsGroupsNames = _.memoize((schema) => {
```

Comment:
All the functions here are well tested indirectly through the exported function in this file. I did this to avoid testing implementation details since the semantics of the exported fn is not that complicated to test.

Reply:
My motivation for adding tests is so each method can be more easily reviewed (and refactorings can be more easily suggested since they can be tested independently)

Reply:
Added unit tests for these fns